Example/WalletApp/PresentationLayer/Wallet/SessionRequest/SessionRequestInteractor.swift

```diff
@@ -3,12 +3,16 @@ import Web3Wallet
 
 final class SessionRequestInteractor {
     func approve(sessionRequest: Request) async throws {
-        let result = Signer.sign(request: sessionRequest)
-        try await Web3Wallet.instance.respond(
-            topic: sessionRequest.topic,
-            requestId: sessionRequest.id,
-            response: .response(result)
-        )
+        do {
+            let result = try Signer.sign(request: sessionRequest)
+            try await Web3Wallet.instance.respond(
+                topic: sessionRequest.topic,
+                requestId: sessionRequest.id,
+                response: .response(result)
+            )
+        } catch {
+            throw error
+        }
```

Comment:
what do we need do catch block here?

Reply:
Just to pass an error on the top level for presenting it with alert on UI

Reply:
but wouldn't it work the same way if it just throws in line 7 or 8?